File: sol/2176/2176.py
from collections import defaultdict

class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        locs = defaultdict(list)
        res = 0
        for j, x in enumerate(nums):
            for i in locs[x]:
                if i * j % k == 0:
                    res += 1
            locs[x].append(j)
        return res

File: sol/2176/2176.cpp
#include <unordered_map>

struct Solution {
    int countPairs(vector<int>& nums, int k) {
        unordered_map<int, vector<int>> locs;
        int res = 0;
        for (int j = 0; j < nums.size(); ++j){
            for (int i : locs[nums[j]]){
                if (i * j % k == 0){
                    ++res;
                }
            }
            locs[nums[j]].push_back(j);
        }
        return res;
    }
};
